### PR TITLE
feat(vpn): add sdk for datasource vpn_customer_gateway

### PR DESCRIPTION
**Pull request by dangyy0626**
**What this PR does / why we need it**:
add sdk for datasource vpn_customer_gateway to search resource

**Special notes for your reviewer**:

**Release note**:

```release-note
api: https://support.huaweicloud.com/api-vpn/vpn_api_0023.html
```

### CODE REVIEW

File: openstack/vpn/v5/customer_gateways/request.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package customer_gateways
+
+import (
+	"github.com/chnsz/golangsdk"
+	"github.com/chnsz/golangsdk/pagination"
+)
+
+func ListAll(c *golangsdk.ServiceClient, opts *ListOpts) ([]CustomerGateway, error) {
```
Comment (ShiChangkuo) [inline openstack/vpn/v5/customer_gateways/request.go:8]: if the function named **ListAll**, why we need ListOpts as a parameter?

File: openstack/vpn/v5/customer_gateways/result.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+type CustomerGateway struct {
+	ID            string `json:"id"`
+	Name          string `json:"name"`
+	IDType        string `json:"id_type,omitempty"`
+	IDValue       string `json:"id_value,omitempty"`
+	BGPAsn        int64  `json:"bgp_asn,omitempty"`
+	IP            string `json:"ip,omitempty"`
+	RouteMode     string `json:"route_mode,omitempty"`
+	CaCertificate CaCertificate
```
Comment (ShiChangkuo) [inline openstack/vpn/v5/customer_gateways/result.go:13]: json tag is missing

File: openstack/vpn/v5/customer_gateways/result.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
+package customer_gateways
+
+import "github.com/chnsz/golangsdk/pagination"
+
+type CustomerGateway struct {
+	ID            string `json:"id"`
+	Name          string `json:"name"`
+	IDType        string `json:"id_type,omitempty"`
```
Comment (ShiChangkuo) [inline openstack/vpn/v5/customer_gateways/result.go:8]: `omitempty` should be removed in a response struct.

File: openstack/vpn/v5/customer_gateways/result.go [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+	PageInfo         pageInfo          `json:"page_info"`
+	TotalCount       int64             `json:"total_count"`
+}
+
+type pageInfo struct {
+	NextMarker   string `json:"next_marker"`
+	CurrentCount int    `json:"current_count"`
+}
+
+type ListOpts struct {
```
Comment (ShiChangkuo) [inline openstack/vpn/v5/customer_gateways/result.go:39]: it better to define it in request.go

File: openstack/vpn/v5/customer_gateways/request.go [hunk 5]
```diff
@@ -0,0 +1,26 @@
+package customer_gateways
```
Comment (ShiChangkuo) [inline openstack/vpn/v5/customer_gateways/request.go:1]: the file name should be request**s**.go and result**s**.go

File: openstack/vpn/v5/customer_gateways/request.go (hunk 1 above)
Comment (dangyy0626) [reply openstack/vpn/v5/customer_gateways/request.go]: removed

**Pull request merged**